File: common_book/src/qbzsydq/java/com/intelligent/reader/adapter/BookShelfItemHolder.java
package com.intelligent.reader.adapter;

import com.bumptech.glide.Glide;
import com.bumptech.glide.load.engine.DiskCacheStrategy;
import com.ding.basic.bean.Book;
import com.intelligent.reader.R;
import com.intelligent.reader.adapter.holder.AbsRecyclerViewHolder;

import net.lzbook.kit.constants.ReplaceConstants;
import net.lzbook.kit.utils.AppUtils;
import net.lzbook.kit.utils.Tools;

import android.text.TextUtils;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

import java.text.MessageFormat;

/**
 * Created by Administrator on 2017/4/13 0013.
 */

public class BookShelfItemHolder extends AbsRecyclerViewHolder<Book> {

    private final ImageView mCheck;
    private final ImageView mImage;
    private final TextView mName, mUpdateTime, mUnread, mLastChapter;
    private final View mFinish, mUpdate, mUnreadLayout;

    public BookShelfItemHolder(View itemView,
                               BookShelfReAdapter.ShelfItemClickListener shelfItemClickListener,
                               BookShelfReAdapter.ShelfItemLongClickListener shelfItemLongClickListener) {
        super(itemView, shelfItemClickListener, shelfItemLongClickListener);

        mCheck = (ImageView) itemView.findViewById(R.id.check_delete);
        mImage = (ImageView) itemView.findViewById(R.id.book_shelf_image);

        mFinish = itemView.findViewById(R.id.book_shelf_status_finish);
        mUpdate = itemView.findViewById(R.id.book_shelf_status_update);
        mName = (TextView) itemView.findViewById(R.id.book_shelf_name);
        mUpdateTime = (TextView) itemView.findViewById(R.id.book_shelf_update_time);
        mLastChapter = (TextView) itemView.findViewById(R.id.book_shelf_last_chapter);
        mUnread = (TextView) itemView.findViewById(R.id.book_shelf_unread);
        mUnreadLayout = itemView.findViewById(R.id.rl_unread);

    }

    @Override
    public void onBindData(int position, Book book, boolean update, boolean isRemoveMode, boolean removeMark) {
        if (!TextUtils.isEmpty(book.getName())) {
            this.mName.setText(book.getName());
        }
        this.mUnreadLayout.setVisibility(View.GONE);

        if (book.getLast_chapter() != null) {

            int count;

            if (book.getSequence() <= 0) {
                count = book.getChapter_count();
            } else {
                count = book.getChapter_count() - (book.getSequence() + 1);
            }

            if (count > 0) {
                this.mUnreadLayout.setVisibility(View.VISIBLE);
                this.mUnread.setText(MessageFormat.format("{0}章", count));
            }

            this.mLastChapter.setText(book.getLast_chapter().getName());

            if (this.mUpdateTime != null) {
                this.mUpdateTime.setText(MessageFormat.format("{0}更新", Tools.compareTime(AppUtils.formatter, book.getLast_chapter().getUpdate_time())));
            }
        }

        if ("FINISH".equals(book.getStatus())) {
            this.mFinish.setVisibility(View.VISIBLE);
        } else {
            this.mFinish.setVisibility(View.GONE);
        }

        if (!update) {
            this.mUpdate.setVisibility(View.GONE);
        } else {
            this.mFinish.setVisibility(View.GONE);
            this.mUpdate.setVisibility(View.VISIBLE);
        }

        if (!TextUtils.isEmpty(book.getImg_url()) && !book.getImg_url().equals(ReplaceConstants.getReplaceConstants().DEFAULT_IMAGE_URL)) {
            Glide.with(itemView.getContext().getApplicationContext()).load(book.getImg_url()).placeholder(R.drawable.icon_book_cover_default).error((R.drawable.icon_book_cover_default)).diskCacheStrategy(DiskCacheStrategy.ALL).into(this.mImage);
        } else {
            Glide.with(itemView.getContext().getApplicationContext()).load(R.drawable.icon_book_cover_default).into(this.mImage);
        }

        if (isRemoveMode) {
            this.mCheck.setVisibility(View.VISIBLE);
            this.mFinish.setVisibility(View.GONE);
            this.mUpdate.setVisibility(View.GONE);
            if (removeMark) {
                this.mCheck.setBackgroundResource(R.mipmap.bookshelf_delete_checked);
            } else {
                this.mCheck.setBackgroundResource(R.mipmap.bookshelf_delete_unchecked);
            }
        } else {
            this.mCheck.setVisibility(View.GONE);
        }
    }
}

File: bookshelf/src/qbmfxsydq/java/com.dingyue.bookshelf/BookShelfItemHolder.kt
package com.dingyue.bookshelf

import android.support.v7.widget.RecyclerView
import android.text.TextUtils
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.bumptech.glide.Glide
import com.bumptech.glide.load.engine.DiskCacheStrategy
import com.ding.basic.bean.Book
import kotlinx.android.synthetic.qbmfxsydq.item_bookshelf_book.view.*
import net.lzbook.kit.app.base.BaseBookApplication
import net.lzbook.kit.constants.Constants
import net.lzbook.kit.constants.ReplaceConstants
import net.lzbook.kit.utils.AppUtils
import net.lzbook.kit.utils.Tools
import net.lzbook.kit.utils.book.RepairHelp
import java.text.MessageFormat

/**
 * Created by Administrator on 2017/4/13 0013
 */
class BookShelfItemHolder(parent: ViewGroup) : RecyclerView.ViewHolder(
        LayoutInflater.from(parent.context).inflate(R.layout.item_bookshelf_book, parent, false)) {

    fun bind(book: Book, bookshelfItemListener: BookShelfAdapter.BookShelfItemListener,
             contains: Boolean, isRemove: Boolean) = with(itemView) {

        if (!TextUtils.isEmpty(book.name)) {
            txt_book_name.text = book.name
        }

        if (!TextUtils.isEmpty(book.author)) {
            txt_book_author.text = book.author
        }

        if (book.sequence + 1 > book.chapter_count) {
            book.sequence = book.chapter_count - 1
        }

        if (book.sequence >= 0) {
            txt_read_progress.text = MessageFormat.format("{0}/{1}章", book.sequence + 1, book.chapter_count)
        } else {
            txt_read_progress.text = "未读"
        }

        val sp = BaseBookApplication.getGlobalContext().getSharedPreferences(Constants.SHAREDPREFERENCES_KEY, 0)
        if (RepairHelp.isShowFixBtn(context, book.book_id) && sp.getBoolean(book.book_id, true)) {
            img_book_update.visibility = View.VISIBLE
            txt_read_progress.text = "章节已修复至最新"
        } else {
            img_book_update.visibility = View.GONE

            // 是否连载
            if (book.status.equals("FINISH")) {
                img_book_status.visibility = View.VISIBLE
            } else {
                img_book_status.visibility = View.GONE
            }
            // 是否有更新
            if (book.update_status != 1) {
                img_book_update.visibility = View.GONE
            } else {
                img_book_update.visibility = View.VISIBLE
                img_book_status.visibility = View.GONE
            }
        }

        if (txt_book_update_time != null && book.last_chapter != null) {
            txt_book_update_time.text = MessageFormat.format("{0}更新:", Tools.compareTime(AppUtils.formatter, book.last_chapter!!.update_time))
        }

        if (!TextUtils.isEmpty(book.img_url) && !book.img_url.equals(
                        ReplaceConstants.getReplaceConstants().DEFAULT_IMAGE_URL)) {
            Glide.with(itemView.context.applicationContext)
                    .load(book.img_url).placeholder(R.drawable.icon_book_cover_default)
                    .error(R.drawable.icon_book_cover_default)
                    .diskCacheStrategy(DiskCacheStrategy.ALL)
                    .into(img_book_cover)
        } else {
            Glide.with(itemView.context.applicationContext)
                    .load(R.drawable.icon_book_cover_default)
                    .into(img_book_cover)
        }


        //显示最后一个章节
        txt_book_last_chapter.text = book.last_chapter?.name ?: ""


        if (isRemove) {
            img_book_select_state.visibility = View.VISIBLE
            img_book_status.visibility = View.GONE
            img_book_update.visibility = View.GONE
            if (contains) {
                img_book_select_state.setBackgroundResource(R.drawable.bookshelf_delete_check_selected_icon)
            } else {
                img_book_select_state.setBackgroundResource(R.drawable.bookshelf_delete_check_select_icon)
            }
        } else {
            img_book_select_state.visibility = View.GONE
        }

        itemView.setOnClickListener {
            bookshelfItemListener.clickedBookShelfItem(book, adapterPosition)
        }

        itemView.setOnLongClickListener {
            bookshelfItemListener.longClickedBookShelfItem()
        }
    }
}
